Lib/subprocess.py

```diff
@@ -1307,15 +1307,18 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
+                    # The encoding here should match the encoding
+                    # written in by the subprocess implementations
+                    # like _posixsubprocess
+                    err_msg = err_msg.decode()
                 except ValueError:
                     exception_name = b'SubprocessError'
                     hex_errno = b'0'
-                    err_msg = (b'Bad exception data from child: ' +
-                               repr(errpipe_data))
+                    err_msg = "Bad exception data from child: {!s}".format(
+                                  bytes(errpipe_data))
```

Comment:
> I asked you twice to keep the repr(), you didn't do it. Why? Can you at least explain?

@ammaraskar, without the repr() the code emits a bytes warning when run Python with the `-b` option.

> I don't understand why you cast a bytes string to bytes: bytes(errpipe_data).

This is explained in the comment by @gpshead.